linkml_runtime/utils/schemaview.py

```diff
@@ -586,6 +590,30 @@ def slot_leaves(self, imports=True, mixins=True) -> List[SlotDefinitionName]:
                 for c in self.all_slots(imports=imports)
                 if self.slot_children(c, mixins=mixins, imports=imports) == []]
 
+    def is_mixin(self, element_name: Union[ElementName, Element]) -> bool:
+        if isinstance(element_name, Element):
+            return element_name
+        else:
+            element = self.get_element(element_name)
+            is_mixin = element.mixin if isinstance(element, Definition) else False
+            return is_mixin
+
+    @lru_cache(CACHE_SIZE)
+    def inverse(self, name: Union[ElementName, Element]) -> str:
+        """
+        Parameters
+        ----------
+        name: str
+            The name or alias of an element
+
+        Returns
+        -------
+        bool
+            true if the element has an inverse
+        """
+        element = self.get_element(name)
+        inverse = element.inverse if isinstance(element, SlotDefinition) else "primary direction"
```

Comment:
this will return "primary direction" if the element is a class or enum

Maybe restrict the signature to be SlotName or SlotDefinition

In general I would also avoid hardcoding strings...

For slots I think there are 3 cases:

 1. asserted inverse
 2. inferred inverse (e.g. `next(inv.name for inv in view.slots if inv.inverse == s.name)`)
 3. no inverse

The concept of 'primary direction' is biolink specific, although we could push this convention to the linkml level, it may make more sense to have the runtime be more generic and concepts of primary direction be at the application level